languages/python/django-oso/tests/test_django_oso.py

```diff
@@ -187,9 +200,18 @@ def test_partial_subfield_isa():
     authorize_filter = authorize_model(None, Post, actor="foo", action="bar")
     assert (
         str(authorize_filter)
-        == "(OR: (AND: ('pk__in', []), ('created_by__name', 'not alice')), ('created_by__name', 'alice'))"
+        == "(OR:"
+        + " (AND: (NOT (AND: ('pk__in', []))), ('pk__in', []), ('created_by__name', 'not alice')),"
```

Comment:
This is a `AND: True, False`? That seems contradictory.  Also I don't see these translated in the SQL.

Reply:
Yeah, the entire thing is:

```
OR(
	AND(true, false, created_by__name="not alice"),
	AND(true, true, created_by__name="alice")
)
```
The Django optimizer removes the first disjunct b/c `AND(true, false, ...)` cannot be `true`. It also optimizes away both `true`s from the second disjunct b/c `AND(true, anything)` is the same as `AND(anything)`.